### PR TITLE
[Terms3] Add `Sum.ops` and `Sum.coeffs` and directly deprecated them

### PR DESCRIPTION
**Pull request by Qottmann**
Following https://github.com/PennyLaneAI/pennylane/pull/5132/ and https://github.com/PennyLaneAI/pennylane/pull/5133, the overarching goal is to achieve feature parity with `qml.Hamiltonian`, which on the other hand is part of the overarching goal of activating new opmath by default.

While for `qml.Hamiltonian` it made sense to access `coeffs` and `ops` directly, in `Sum` and `Prod` it requires some processing so it makes more sense (and costs less) to directly call the `terms()` method to obtain both.

[sc-56284]
[sc-53963]

### CODE REVIEW

**Comment by Qottmann** [comment]
> What is the additional processing? Aren't we just doing coefs, ops = Sum.terms()

Yes! What I meant was that in `Hamiltonian`, `H.ops` and `H.coeffs` (as well as `H.terms()`) are directly accessed because they define the object, whereas in the case of `Sum` we need to do some processing (namely the one in `Sum.terms()`), so accessing for example `Sum.ops` and `Sum.coeffs` calls the `terms()` method twice, which is why we suggest users to use `Sum.terms()` in the first place to avoid that culprit.

File: pennylane/ops/op_math/prod.py [hunk 1]
```diff
@@ -472,6 +473,26 @@ def terms(self):
                 ops.append(factor)
         return coeffs, ops
 
+    @property
+    def coeffs(self):
+        r"""``coeffs`` in ``coeffs, ops = op.terms()``"""
```
Comment (albi3ro) [inline pennylane/ops/op_math/prod.py:478]: Might be worth fleshing out these docstrings a little more, and point the user to both `Operator.terms` and `Operator.pauli_rep`.

Comment (Qottmann) [reply pennylane/ops/op_math/prod.py]: Good point ✅

**Review by albi3ro — APPROVED** [review]
👍

**Review by PietropaoloFrisoni — APPROVED** [review]
Looks great!

**Pull request merged**